app/services/return-logs/setup/submit-readings.service.js

```diff
@@ -0,0 +1,124 @@
+'use strict'
+
+/**
+ * Orchestrates validating the data for `/return-logs/setup/{sessionId}/readings/{yearMonth}` page
+ * @module SubmitReadingsService
+ */
+
+const GeneralLib = require('../../../lib/general.lib.js')
+const ReadingsPresenter = require('../../../presenters/return-logs/setup/readings.presenter.js')
+const ReadingsValidator = require('../../../validators/return-logs/setup/readings.validator.js')
+const SessionModel = require('../../../models/session.model.js')
+
+/**
+ * Orchestrates validating the data for `/return-logs/setup/{sessionId}/readings/{yearMonth}` page
+ *
+ * @param {string} sessionId - The UUID of the current session
+ * @param {object} payload - The submitted form data
+ * @param {object} yar - The Hapi `request.yar` session manager passed on by the controller
+ * @param {string} yearMonth - The year and zero-indexed month to view, eg. `2014-0` for January 2014
+ *
+ * @returns {Promise<object>} If no errors it returns an empty object else the page data for the readings page including
+ * the validation error details
+ */
+async function go(sessionId, payload, yar, yearMonth) {
+  const session = await SessionModel.query().findById(sessionId)
+
+  const [requestedYear, requestedMonth] = _determineRequestedYearAndMonth(yearMonth)
+
+  const validationResult = _validate(payload, requestedYear, requestedMonth, session)
+
+  if (!validationResult) {
+    await _save(payload, session, requestedYear, requestedMonth)
+
+    GeneralLib.flashNotification(yar, 'Updated', 'Readings have been updated')
+
+    return {}
+  }
+
+  _addValidationResultToSession(payload, session, requestedYear, requestedMonth, validationResult)
+
+  const formattedData = ReadingsPresenter.go(session, yearMonth)
+
+  return {
+    activeNavBar: 'search',
+    error: validationResult,
+    ...formattedData
+  }
+}
+
+function _addValidationResultToSession(payload, session, requestedYear, requestedMonth, validationResult) {
+  session.lines.forEach((line) => {
+    const endDate = new Date(line.endDate)
+
+    if (endDate.getFullYear() === requestedYear && endDate.getMonth() === requestedMonth) {
+      // Unlike when the session is saved, we do not convert the reading to a number here. This is because we want to
+      // return what was submitted in the payload to the view following failed validation, which could be a string
+      line.reading = payload[line.endDate] ? payload[line.endDate] : null
```

Comment:
Could you use the nullish coalescing here
```suggestion
     line.reading = payload[line.endDate] ?? null
```